components/nexus-repository/src/main/java/org/sonatype/nexus/repository/storage/StorageFacetImpl.java

```diff
@@ -85,6 +96,16 @@ protected void doInit() throws Exception {
     super.doInit();
   }
 
+  @Override
+  protected void doStart() throws Exception {
+    try {
+      searchFacet = getRepository().facet(SearchFacet.class);
+    }
+    catch (MissingFacetException e) {
+      // no search facet, no indexing
```

Comment:
are there real cases where a repo would not have a search facet?  ie is the nullability here causing additional logic/code to manage a situation that likely will never happen?